### PR TITLE
Add EL9 support

### PR DESCRIPTION
**Pull request by bastelfreak**
## Summary
Provide a detailed description of all the changes present in this pull request.

## Additional Context
Add any additional context about the problem here. 
- [ ] Root cause and the steps to reproduce. (If applicable)
- [ ] Thought process behind the implementation.

## Related Issues (if any)
Mention any related issues or pull requests.

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified. (For example `puppet apply`)

### CODE REVIEW

File: spec/acceptance/server_instance_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # run a test task
 require 'spec_helper_acceptance'
 
-describe 'postgresql instance test1', if: os[:family] == 'redhat' && os[:release].start_with?('8') do
+describe 'postgresql instance test1', if: os[:family] == 'redhat' && os[:release].split.first.to_i >= 8 do
```
Comment (ekohl) [inline spec/acceptance/server_instance_spec.rb:6]: I'd just exclude EL7. IMHO easier to read.
```suggestion
describe 'postgresql instance test1', if: os[:family] == 'redhat' && !os[:release].start_with?('7') do
```

**Review by ekohl — APPROVED** [review]

**Comment by SimonHoenscheid** [comment]
EL9 manages the default postgres in dnf modules, the postgres repo does not. 

Recommended way is to disable the postgres dnf modules and install from the postgres repo. 

the postgres repo requires a version which is defined via postgresql::globals::version, this is also the default version which is handed to the dnf module, ehich would need disabled here, to act correcty, the globals class requires a patch to make this possible.

My suggestion is to add an optional_parameter like dnf_module_version, unless set real_dnf_module_version defaults to $version/global_version.

https://github.com/puppetlabs/puppetlabs-postgresql/blob/main/manifests/globals.pp#L289

File: spec/acceptance/server_instance_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # run a test task
 require 'spec_helper_acceptance'
 
-describe 'postgresql instance test1', if: os[:family] == 'redhat' && os[:release].start_with?('8') do
+describe 'postgresql instance test1', if: os[:family] == 'redhat' && !os[:release].start_with?('7') do
   pp = <<-MANIFEST
   # set global defaults
   class { 'postgresql::globals':
```
Comment (ekohl) [inline spec/acceptance/server_instance_spec.rb:9]: I'd change the version to 15 or 16. I don't have a RHEL 8 at hand to see if it ships 16 already, so I'd suggest 15.

**Review by ekohl — CHANGES_REQUESTED** [review]
> EL9 manages the default postgres in dnf modules, the postgres repo does not.

For plain EL9 this isn't true:
```console
# dnf install postgresql
...
# dnf module list --enabled -q
```

You can install PostgreSQL 15 or 16 via modules, but the default (13) is not modular.

The reason the test fails is that it sets `manage_dnf_module` to `true` and `version` to `'13'`. That isn't a combination that exists.

The easiest fix is to pick a version that exists in both EL 8 & 9 as modular versions, which means picking 15 or 16.

File: spec/acceptance/server_instance_spec.rb (hunk 2 above)
Comment (vchepkov) [reply spec/acceptance/server_instance_spec.rb:9]:
```
Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
Name                                                                     Stream                                                               Profiles                                                                        Summary                                                                                         
postgresql                                                               9.6                                                                  client, server [d]                                                              PostgreSQL server and client module                                                             
postgresql                                                               10 [d]                                                               client, server [d]                                                              PostgreSQL server and client module                                                             
postgresql                                                               12                                                                   client, server [d]                                                              PostgreSQL server and client module                                                             
postgresql                                                               13                                                                   client, server [d]                                                              PostgreSQL server and client module                                                             
postgresql                                                               15                                                                   client, server [d]                                                              PostgreSQL server and client module                                                             
postgresql                                                               16                                                                   client, server [d]                                                              PostgreSQL server and client module                                             
```

Comment (ekohl) [reply spec/acceptance/server_instance_spec.rb:9]: @bastelfreak could you change line 15 from `'13'` to `'16'`? I'd then expect it to pass.

Comment (bastelfreak) [reply spec/acceptance/server_instance_spec.rb:9]: done, let's see.

**Review by ekohl — APPROVED** [review]
Thanks

**Pull request merged**